### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by silviatofana**
The following were implemented in this milestone:

Create a class Classroom with the following:
. @Label instance variable, should be initialized in the constructor.
. Setter and getter for @Label (remember about attr_accessor).
Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
. Create the has-many side (a classroom has many students).
. Create the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
Create a class Book with the following:
. @title and https://github.com/author instance variables, should be initialized in the constructor.
. Setters and getters for instance variables (remember about attr_accessor).
Create a class Rental with the following:
. @Date instance variable, should be initialized in the constructor.
. Setter and getter for @Date (remember about attr_accessor).
Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
. Create the has-many side of Book and Rental (a book has many rentals).
. Create the belongs-to side of Rental and Book (a rental belongs to a book).
. Create the has-many side of Person and Rental (a person has many rentals).
. Create the belongs-to side of Rental and Person (a rental belongs to a person).
. Modify the constructor of Rental so Book and Person are set in it.
Please review my code.
Thanks!

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
   attr_reader :id
 
-  def initialize(age:, name: 'Unknown', parent_permission: true)
+  def initialize(age, name: 'Unknown', parent_permission: true)
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (omarsalem7) [inline person.rb:16]: - [ ] kindly add `has-many` relation between rental and person. By adding a method `add_rental` in Person class like you did before in Book class.
>image

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @silviatofana 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Descriptive pull request ✔️
- Correct Git-flow ✔️
- Good descriptive README ✔️

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @silviatofana, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by silviatofana** [comment]
thanks

**Pull request merged**